onnxruntime/test/providers/cpu/tensor/pad_test.cc

```diff
@@ -809,22 +809,68 @@ TEST(PadOpTest, ConstantPadAxes) {
+                          1, 1});
   test.AddInput<int64_t>("pads", {4}, {0, 1, 0, 1});
   test.AddInput<int32_t>("value", {1}, {0});
   test.AddInput<int32_t>("axes", {2}, {1, 3});
   test.AddOutput<int32_t>("output", {1, 2, 2, 4},
-  {
-    0, 1, 1, 0,
-    0, 1, 1, 0,
-    0, 1, 1, 0,
-    0, 1, 1, 0
-    }
-  );
+                          {0, 1, 1, 0,
+                           0, 1, 1, 0,
+                           0, 1, 1, 0,
+                           0, 1, 1, 0});
+  // CoreML EP requires axes if provided to be a constant initializer and only supports padding specified on last two
+  // dimensions, skip this test for CoreML EP here.
+  test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider, kCoreMLExecutionProvider});
+}
+
+TEST(PadOpTest, ConstantPadAxes_1) {
+  // CoreML EP only supports padding on last two dimensions and requires axes to be an initializer if provided,
+  // added the following test case (can be taken by CoreML):
+  // Given a provided axes input with last two dimensions specified.
+  OpTester test("Pad", 18);
+  test.AddAttribute("mode", "constant");
+  test.AddInput<int32_t>("data", {1, 2, 2, 2},
+                         {1, 1,
+                          1, 1,
+                          1, 1,
+                          1, 1});
+  test.AddInput<int64_t>("pads", {4}, {0, 1, 0, 1});
+  test.AddInput<int32_t>("value", {1}, {0});
+  test.AddInput<int64_t>("axes", {2}, {2, 3}, true /* axes_is_initializer */);
+  test.AddOutput<int32_t>("output", {1, 2, 2, 4},
+                          {0, 1, 1, 0,
+                           0, 1, 1, 0,
+                           0, 1, 1, 0,
+                           0, 1, 1, 0});
+  test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider});
+}
+
+TEST(PadOpTest, ConstantPadAxes_2) {
+  // CoreML EP only supports padding on last two dimensions and requires axes to be an initializer if provided,
+  // added the following test case (can be taken by CoreML):
+  // Given a provided axes input with last two dimensions specified.
+  OpTester test("Pad", 18);
+  test.AddAttribute("mode", "constant");
+  test.AddInput<int32_t>("data", {1, 2, 2, 2},
+                         {1, 1,
+                          1, 1,
+                          1, 1,
+                          1, 1});
+  test.AddInput<int64_t>("pads", {4}, {1, 1, 1, 1});
+  test.AddInput<int32_t>("value", {1}, {0});
+  test.AddInput<int64_t>("axes", {2}, {2, 3}, true /* axes_is_initializer */);
+  test.AddOutput<int32_t>("output", {1, 2, 4, 4},
+                          {0, 0, 0, 0,
+                           0, 1, 1, 0,
+                           0, 1, 1, 0,
+                           0, 0, 0, 0,
+                           0, 0, 0, 0,
+                           0, 1, 1, 0,
+                           0, 1, 1, 0,
+                           0, 0, 0, 0});
   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider});
 }
```

Comment:
some other test ideas:
- negative axes
- out of order axes values
- axes with 0's in the pad values for dimensions other than the last two
- specifying padding for only one dimension of the last two

I agree with @skottmckay's comment that the test names could be more descriptive.